Objects/listobject.c

```diff
@@ -2557,6 +2557,27 @@ PyList_AsTuple(PyObject *v)
     return _PyTuple_FromArray(((PyListObject *)v)->ob_item, Py_SIZE(v));
 }
 
+PyObject *
+_PyList_FromArraySteal(PyObject *const *src, Py_ssize_t n)
+{
+    if (n == 0) {
```

Comment:
Why special case `0`?

Reply:
Presumably copied from `_PyTuple_FromArraySteal`, where the special case makes sense (since only the empty tuple can be freely shared).

Here it still *might* make sense since `[]` is probably the most common list constructor. 

Also, is memcpy() with a zero count well-defined?

Reply:
`memcpy()` with a zero count is well defined (C11 7.23.1.2), however it requires that the destination pointer be valid. Since `PyList_New(0)` sets `ob_item` to `NULL`, `memcpy()` can't be called -- at all -- for empty lists.